metricbeat/module/jolokia/jmx/jmx.go

```diff
@@ -54,76 +53,55 @@ var (
 // MetricSet type defines all fields of the MetricSet
 type MetricSet struct {
 	mb.BaseMetricSet
-	mapping   AttributeMapping
+	mapping   []JMXMapping
 	namespace string
-	http      *helper.HTTP
+	http      JolokiaHTTPRequestBuilder
 	log       *logp.Logger
 }
 
 // New create a new instance of the MetricSet
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	config := struct {
-		Namespace string       `config:"namespace" validate:"required"`
-		Mappings  []JMXMapping `config:"jmx.mappings" validate:"required"`
+		Namespace  string       `config:"namespace" validate:"required"`
+		HTTPMethod string       `config:"http_method"`
+		Mappings   []JMXMapping `config:"jmx.mappings" validate:"required"`
 	}{}
 
 	if err := base.Module().UnpackConfig(&config); err != nil {
 		return nil, err
 	}
 
-	body, mapping, err := buildRequestBodyAndMapping(config.Mappings)
-	if err != nil {
-		return nil, err
-	}
-
-	http, err := helper.NewHTTP(base)
-	if err != nil {
-		return nil, err
-	}
-	http.SetMethod("POST")
-	http.SetBody(body)
+	jolokiaHTTPBuild := NewJolokiaHTTPRequestBuiler(config.HTTPMethod)
```

Comment:
This is not a builder anymore, it is a Fetcher now :slightly_smiling_face:

Reply:
Hi @jsoriano ! Thanks for noticing! Renamed to interface to `JolokiaHTTPRequestFetcher` and structs to `JolokiaHTTPGetFetcher` and `JolokiaHTTPPostFetcher`. :+1: